src/uucore/src/lib/features/format/mod.rs

```diff
@@ -308,20 +308,21 @@ pub fn sprintf<'a>(
     Ok(writer)
 }
 
-/// A parsed format for a single float value
+/// A parsed format for a single numerical value of type T
 ///
-/// This is used by `seq`. It can be constructed with [`Format::parse`]
+/// This is used by `seq` and `csplit`. It can be constructed with [`Format::parse`]
 /// and can write a value with [`Format::fmt`].
 ///
 /// It can only accept a single specification without any asterisk parameters.
 /// If it does get more specifications, it will return an error.
-pub struct Format<F: Formatter> {
+pub struct Format<F: Formatter<T>, T> {
     prefix: Vec<u8>,
     suffix: Vec<u8>,
     formatter: F,
+    _marker: std::marker::PhantomData<T>,
```

Comment:
This `PhantomData` thing doesn't make me happy, but I couldn't find another way to have a `Formatter` that either takes a value parameter (`i64`/`u64`) _or_ a reference (`&ExtendedBigDecimal`).

Ideas welcome....